perfkitbenchmarker/providers/azure/azure_blob_storage.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import linux_packages
+from perfkitbenchmarker import object_storage_service
+from perfkitbenchmarker import providers
+from perfkitbenchmarker import vm_util
+
+flags.DEFINE_string('azure_lib_version', None,
+                    'Use a particular version of azure client lib, e.g.: 1.0.2')
+
+FLAGS = flags.FLAGS
+
+DEFAULT_AZURE_REGION = 'East US'
+AZURE_CREDENTIAL_LOCATION = '.azure'
+
+
+class AzureBlobStorageService(object_storage_service.ObjectStorageService):
+  """Interface to Azure Blob Storage."""
+
+  STORAGE_NAME = providers.AZURE
+
+  def PrepareService(self, location):
+    self.storage_account = 'pkb%s' % FLAGS.run_uri
+    vm_util.IssueCommand(
+        ['azure', 'storage', 'account', 'create',
+         '--type', 'ZRS',
+         '-l', location or DEFAULT_AZURE_REGION,
+         self.storage_account])
+
+    output, _, _ = vm_util.IssueCommand(
+        ['azure', 'storage', 'account',
+         'keys', 'list', self.storage_account])
+
+    key = re.findall(r'Primary:* (.+)', output)
+    self.azure_key = key[0]
+
+  def CleanupService(self):
+    vm_util.IssueCommand(
+        ['azure', 'storage', 'account', 'delete',
+         '-q', self.storage_account])
+
+  def MakeBucket(self, bucket):
+    vm_util.IssueCommand(
+        ['azure', 'storage', 'container',
+         'create', bucket,
+         '-a', self.storage_account,
+         '-k', self.azure_key])
+
+  def DeleteBucket(self, bucket):
+    vm_util.IssueCommand(
+        ['azure', 'storage', 'container',
+         'delete', '-q', bucket,
+         '-a', self.storage_account,
+         '-k', self.azure_key])
+
+  def EmptyBucket(self, bucket):
+    # Emptying buckets on Azure is hard. We pass for now - this will
+    # increase our use of storage space, but should not affect the
+    # benchmark results.
+    pass
```

Comment:
Thanks for the explanation. My concern is we left blobs and buckets behind, but since we can still delete the bucket with blobs inside, it is fine.

Correct me if I am wrong, current behavior is during CLI tests, GCS and S3 remove all objects after each round of uploads. But on Azure we overwrite the objects? Which still seems a bit inconsistent to me.

Consider adding a TODO, we should support this method eventually (when Azure supports recursive/wildcard deletes) or maybe consider invoke api delete script if possible.